cmd/tempo/app/server_service.go

```diff
@@ -3,14 +3,101 @@ package app
 import (
 	"context"
 	"fmt"
+	"net/http"
+	"strings"
 
+	"github.com/go-kit/log"
 	"github.com/go-kit/log/level"
+	"github.com/gorilla/mux"
+	"github.com/grafana/dskit/middleware"
 	"github.com/grafana/dskit/server"
 	"github.com/grafana/dskit/services"
+	"golang.org/x/net/http2"
+	"golang.org/x/net/http2/h2c"
+	"google.golang.org/grpc"
 
 	util_log "github.com/grafana/tempo/pkg/util/log"
 )
 
+type TempoServer interface {
```

Comment:
Do we need the interface?  It looks like we would be able to use the struct directly.  Are you anticipating adding another implementation?

Reply:
the primary reason i used an interface was to limit the ways in which the rest of the code could interact with the `t.server` var. Another option would have been to move this into its own package but i suppose that didn't feel necessary.

Reply:
I suppose my feeling is that interfaces should have more than one implementation, but I see your perspective to limit the interactions.